### PR TITLE
54 supporting other lidar interpolation options

### PR DESCRIPTION
**Pull request by rosepearson**
Add support for calculating the mean, median or IDW of LiDAR points during the DEM generation process:
 - [x] Make code change
 - [x] Update tests 
   - [x] Update / create new tests
   - [x] Ensure these tests have the expected behaviour
   - [x] Test locally and ensure tests are passing
 - [x] Ensure tests passing on GH Actions
 - [x] Update documentation
   - [x] Doc strings
   - [x] Wiki 
  - [x] Update package version (if needed)

### CODE REVIEW

**Comment by rosepearson** [comment]
Hi @Martin20494,
I've implemented support for calculating the mean, median or IDW from LiDAR points during the DEM generation process. I've done this by adding a new instruction file parameter `'lidar_interpolation_method` with accepted values of 'idw', 'mean' and 'median'.

Do you mind reviewing my PR and accepting these changes if you are happy with them? I'll create a meeting for next Tuesday to run through what's involved in performing a PR.

**Comment by Martin20494** [comment]
Hi @rosepearson,
That's awesome! Thank you so much Rose for doing this. I might need more than a week to have a try with it as I am in between of writing some reports, but will try my best! I think we also need min and max, but I will discuss with Matt about this.

**Comment by rosepearson** [comment]
Awesome @Martin20494 RE Min and Max. I was wondering. I'll add them now. Let me know if there is any other averaging you may want too and I can add them prior to our meeting tomorrow.

Rose

**Comment by Martin20494** [comment]
Awesome! Thank you Rose @rosepearson

File: src/geofabrics/dem.py [hunk 1]
```diff
@@ -886,34 +887,63 @@ def read_file_with_pdal(lidar_file: typing.Union[str, pathlib.Path], region_to_t
...
     for i, (near_indicies, point) in enumerate(zip(tree_index_list, xy_out)):
 
         if len(near_indicies) == 0:  # Set NaN if no values in search region
             z_out[i] = numpy.nan
         else:
-            distance_vectors = point - tree.data[near_indicies]
-            smoothed_distances = numpy.sqrt(((distance_vectors**2).sum(axis=1)+smoothing**2))
-            if smoothed_distances.min() == 0:  # in the case of an exact match
-                z_out[i] = point_cloud['Z'][tree.query(point, k=1)[1]]
+            if options['method'] == 'mean':
```
Comment (rosepearson) [inline src/geofabrics/dem.py:908]: This is where support has been added for the new interpolation methods.

**Review by rosepearson — COMMENTED** [review]
I have spoken to @Martin20494 and he is happy with this as a first go at adding support for basic interpolation methods. He mentioned we may want to support another method from a paper in the future.

File: src/geofabrics/dem.py (hunk 1 above)
Comment (rosepearson) [reply src/geofabrics/dem.py:908]: @Martin20494 noted these options are also available in PDAL.

**Pull request merged**